code/pilotfile/plr.cpp

```diff
@@ -728,9 +737,51 @@ void pilotfile::plr_read_controls()
 void pilotfile::plr_write_controls()
 {
 	handler->startSectionWrite(Section::Controls);
-	
+
+	// Save the default bindings to prevent crash. See github issue 3902
+	auto& Control_config_default = Control_config_presets[0].bindings;
```

Comment:
Couldn't this just be the current preset instead, since it's referenced just below here already? That would allow for a basically consistent control setup between the builds. And it would also make that "technical difficulties" info box pretty much irrelevant.

Reply:
No. It cannot be the current preset.  This is because the new binding scheme allows for multiple joysticks and the mouse to be bound to the same control, and the old binding scheme can not achieve that.  When the compatibility code goes to convert the post-Controls5 bindings into pre-Controls5 bindings, it loses the multiple joysticks (and mouse) bindings making it significantly different from the preset it was based on.  Thus, when the compatibility portion of the pilot is re-imported, the plr conversion code sees it as a completely new preset and essentially duplicates the bindings.

Reply:
Ah, right.